src/ext/serial.cpp

```diff
@@ -65,24 +118,41 @@ template<typename T> class ThingSetSerial: public ThingSetStream
  */
 
 #ifdef UART_SERIAL_ENABLED
-    extern const int pub_channel_serial;
-    extern Serial serial;
 
-    ThingSetSerial ts_uart(serial, pub_channel_serial);
+    #ifdef __MBED__
+        extern const int pub_channel_serial;
+        extern Serial serial;
+        ThingSetSerial ts_uart(serial, pub_channel_serial);
+
+    #elif defined(__ZEPHYR__)
+        extern const int pub_channel_serial;
+        ThingSetSerial ts_uart(pub_channel_serial);
+    #endif /* MBED or ZEPHYR */
+
 #endif /* UART_SERIAL_ENABLED */
 
 // ToDo: Add USB serial again (previous library was broken with recent mbed releases)
 
-char ThingSetStream::buf_resp[1000];
-
 extern ThingSet ts;
 
+int len_ts;
+
 void ThingSetStream::process_1s()
 {
     if (ts.get_pub_channel(channel)->enabled) {
+        #ifdef __MBED__
         ts.pub_msg_json(buf_resp, sizeof(buf_resp), channel);
         stream->puts(buf_resp);
         stream->putc('\n');
+
+        #elif defined(__ZEPHYR__)
+        int len = ts.pub_msg_json(buf_resp, sizeof(buf_resp), channel);
+        len_ts = len;
+
+        uart_irq_rx_disable(uart_dev);
+        uart_irq_tx_enable(uart_dev);
```

Comment:
Why not just call `uart_poll_out` here to send out the response? This is the equivalent of `stream->putc` in mbed. If we handle also the transmission via callbacks and ISR, our characters is mixed with other console output.
I guess it's fine to write all data at once and block this thread for that period of time. Other ext interfaces can wait, humans will not realize if the display is updated a little bit later because we needed to finish sending out data on the serial interface. We just need the callback for RX to make sure we don't miss any characters while doing something else.